### ISSUE TITLE
In MISP versions 2.4.137-140, uploading STIX v1 (XML) via PyMISP will fail

### ISSUE THREAD

**Issue by steves2020** [MISP/PyMISP#711]
In MISP versions 2.4.137-140 uploading STIX v1 (XML) via PyMISP will fail as MISP 2.4.137 and above rejects POSTs with mismatched content-type.

In line 3422 of PyMISP v2.4.140 the content type is hard coded to `application/json`. MISP version 2.4.137 and above seem to now enforce Content Type on requests (see sample code below).

https://github.com/MISP/PyMISP/blob/39a7b8242f0d3022276d417ec334bb46b890ff23/pymisp/api.py#L3422

This behaviour changed at v2.4.130 of PyMISP. 

- v2.4.128 https://github.com/MISP/PyMISP/blob/f176fb0884770ff303de2c4db6606fbe4fcc4199/pymisp/api.py#L2358

- v2.4.130-140   https://github.com/MISP/PyMISP/blob/3b639997853ab3774e1bc19fe95045bf7b2d4f47/pymisp/api.py#L2504

My temp fix (or revert to code in v2.4.128):

       prepped.headers.update(
            {'Authorization': self.key,
             'Accept': f'application/{output_type}',
             'Content-Type': 'application/xml' if output_type=="xml" else 'application/json',
             'User-Agent': user_agent})

Sample code to recreate issue:

    import io
    from pymisp import ExpandedPyMISP
    stix="""<stix:STIX_Package
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xmlns:stix="http://stix.mitre.org/stix-1"
        xmlns:indicator="http://stix.mitre.org/Indicator-2"
        xmlns:cybox="http://cybox.mitre.org/cybox-2"
        xmlns:DomainNameObj="http://cybox.mitre.org/objects#DomainNameObject-1"
        xmlns:cyboxVocabs="http://cybox.mitre.org/default_vocabularies-2"
        xmlns:stixVocabs="http://stix.mitre.org/default_vocabularies-1"
        xmlns:example="http://example.com/"
        xsi:schemaLocation=
        "http://stix.mitre.org/stix-1 ../stix_core.xsd
        http://stix.mitre.org/Indicator-2 ../indicator.xsd
        http://cybox.mitre.org/default_vocabularies-2 ../cybox/cybox_default_vocabularies.xsd
        http://stix.mitre.org/default_vocabularies-1 ../stix_default_vocabularies.xsd
        http://cybox.mitre.org/objects#DomainNameObject-1 ../cybox/objects/Domain_Name_Object.xsd"
        id="example:STIXPackage-f61cd874-494d-4194-a3e6-6b487dbb6d6e"
        timestamp="2014-05-08T09:00:00.000000Z"
        version="1.1.1"
        >
        <stix:STIX_Header>
            <stix:Title>Example watchlist that contains domain information.</stix:Title>
            <stix:Package_Intent xsi:type="stixVocabs:PackageIntentVocab-1.0">Indicators - Watchlist</stix:Package_Intent>
        </stix:STIX_Header>
        <stix:Indicators>
            <stix:Indicator xsi:type="indicator:IndicatorType" id="example:Indicator-2e20c5b2-56fa-46cd-9662-8f199c69d2c9" timestamp="2014-05-08T09:00:00.000000Z">
                <indicator:Type xsi:type="stixVocabs:IndicatorTypeVocab-1.1">Domain Watchlist</indicator:Type>
                <indicator:Description>Sample domain Indicator for this watchlist</indicator:Description>
                <indicator:Observable id="example:Observable-87c9a5bb-d005-4b3e-8081-99f720fad62b">
                    <cybox:Object id="example:Object-12c760ba-cd2c-4f5d-a37d-18212eac7928">
                        <cybox:Properties xsi:type="DomainNameObj:DomainNameObjectType" type="FQDN">
                            <DomainNameObj:Value condition="Equals" apply_condition="ANY">malicious1.example.com##comma##malicious2.example.com##comma##malicious3.example.com</DomainNameObj:Value>
                        </cybox:Properties>
                    </cybox:Object>
                </indicator:Observable>
            </stix:Indicator>
        </stix:Indicators>
    </stix:STIX_Package>
    """
    
    api = ExpandedPyMISP("http://misp", "myapikey")
    stix_fh = io.StringIO(stix)
    response = api.upload_stix(path=stix_fh, version="1")
    print(response.text)

Error message:

    <?xml version="1.0" encoding="UTF-8"?>
    <response><name>Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has been blocked to avoid an unfiltered request.</name><message>Invalid JSON input. Make sure that the JSON input is a correctly formatted JSON string. This request has been blocked to avoid an unfiltered request.</message><url>/events/upload_stix</url></response>

**Comment by Rafiot** [comment]
Wow, congrats for being the first one to notice this bug in 9 months :D 

Fixing it now.

**Comment by Rafiot** [comment]
I didn't package it yet, but can you give a try with the commit I just did?

**Comment by steves2020** [comment]
That works perfectly. Thank you for fixing it so promptly.

```
$ git log | head
commit 0697f1470b25aa15c7406d39c76cb977c957ee25
Author: Raphaël Vinot <raphael@vinot.info>
Date:   Thu Mar 4 12:35:23 2021 +0100

    fix: Re-enable support for uploading STIX 1 documents

$ python test_stix_upload.py
{"Event":{"id":"1156","orgc_id":"66","org_id":"66","date":"2014-05-08","threat_level_id":"3","info":"Example watchlist that contains domain information.",
"published":false,"uuid":"a87809d8-b294-4cd9-ab29-34a0e61793c0","attribute_count":"5","analysis":"0","timestamp":"1614886406","distribution":"1","proposal_email_lock":false,
"user_id":"17","locked":false,"publish_timestamp":"0","sharing_group_id":"0","disable_correlation":false,"extends_uuid":""},"ThreatLevel":{"name":"Low","id":"3"},...
```

**Comment by Rafiot** [comment]
great, thank you for the feedback!
